### DISCUSSION TITLE
Create coc-oxc for vim / nvim

### DISCUSSION THREAD

**Discussion by Boshen** [oxc-project/oxc#3310] [Feature Request]
Should be similar to https://github.com/fannheyward/coc-biome

**Comment by fannheyward** [comment]
I'll try this.

**Comment by Boshen** [comment]
> I'll try this.

Thank you!

**Comment by fannheyward** [comment]
You don't need an extension for now, add the following to coc-settings.json to make oxc server to work:

```json
  "languageserver": {
    "oxc": {
      "command": "oxc_vscode",
      "filetypes": [
        "typescript",
        "javascript"
      ],
      "rootPatterns": [
        ".git"
      ]
    }
  },
```

<img width="772" alt="image" src="https://github.com/oxc-project/oxc/assets/345274/5871edcc-b4da-4f12-b6ff-b7b76a8ae7d7">

**Comment by fannheyward** [comment]
For coc-oxc, maybe something more:

1. restart server if needed
2. fixAll like `eslint`

If oxc supports more off-LSP features, the extension is needed, for now, I think coc-settings.json is enough.

The main reason to create coc-biome is `biome __run_server`, make coc.nvim connect biome with socket, single biome process for multiple projects. You can also use biome without coc-biome:

```json
  "languageserver": {
    "biome": {
      "command": "biome",
      "args": [
        "lsp-proxy"
      ],
      "filetypes": [
        "typescript",
        "javascript"
      ],
      "rootPatterns": [
        ".git"
      ]
    }
  },
```

**Comment by Boshen** [comment]
I see, thank you @fannheyward!

**Comment by fannheyward** [comment]
I'll follow oxc and make coc-oxc in some time. oxc is an excited project, with the incoming prettier feature, all I missing is bundler as esbuild.

**Comment by fannheyward** [comment]
Make https://github.com/oxc-project/oxc/tree/main/editors/vscode/server into a separate crate, something like `oxc_ls`, will be easy to use in editors.

**Comment by rzvxa** [comment]
What is the state of this issue? I'm a vim/neovim user and I love a few good cocs in my editor, is there anything that I can help with? I have experience with both Vim script and Lua but it shouldn't be necessary for this task.

**Reply by Boshen** [reply]
Oh nice we are vim bros. I stopped working on JavaScript entirely so I never took the chance to integrate oxlint into coc :-(

**Reply by rzvxa** [reply]
I see you are a man of culture😆I use javascript for my clients from time to time; I might be able to pick it up, Especially if there is some demand for it.

**Comment by Boshen** [comment]
https://github.com/oxc-project/coc-oxc
